test/e2e/fqdn_dns_cache_test.go

```diff
@@ -0,0 +1,318 @@
+	require.NoError(t, err, "failed to create custom dns ConfigMap: %v", err)
+
+	createCustomDnsPod(t, data)
+
+	// set the custom DNS server IP address in Antrea configMap.
+	setDnsServerAddressInAntrea(t, data, dnsServiceIP)
+	defer setDnsServerAddressInAntrea(t, data, "") //reset after the test.
+
+	createFqdnPolicyInNamespace(t, data, testFQDN)
+	createToolboxPod(t, data, dnsServiceIP)
+
+	curlTarget := func(podName, containerName, fqdn string) (string, error) {
+		cmd := []string{"curl", fqdn}
+		stdout, stderr, err := data.RunCommandFromPod(data.testNamespace, podName, containerName, cmd)
+		if err != nil {
+			return "", fmt.Errorf("error when running command '%s' on Pod '%s': %v, stdout: <%v>, stderr: <%v>",
+				strings.Join(cmd, " "), podName, err, stdout, stderr)
+		}
+		return stdout, nil
+	}
+
+	assert.EventuallyWithT(t, func(collect *assert.CollectT) {
+		stdout, err := curlTarget(toolboxPodName, toolboxContainerName, testFQDN)
+		assert.NoError(t, err)
+		t.Logf("response of curl to FQDN - %s", stdout)
+	}, 2*time.Second, 100*time.Millisecond, "trying to curl the fqdn : ", testFQDN)
+
+	// confirm that the FQDN resolves to the expected IP address and store it to simulate caching of this IP associated with the FQDN.
+	t.Logf("Resolving FQDN to simulate caching the current IP inside Toolbox pod...")
+	resolvedIP, err := data.runDNSQuery(toolboxPodName, toolboxContainerName, data.testNamespace, testFQDN, false, dnsServiceIP)
+	fqdnIp := resolvedIP.String()
+	require.NoError(t, err, "failed to resolve FQDN to an IP from toolbox pod : %v", err)
+	require.Equalf(t, agnHostPodOneIP, fqdnIp, "The IP set against the FQDN in the DNS server should be the same, but got %s instead of %s", fqdnIp, agnHostPodOneIP)
+	t.Logf("Successfully received the expected IP %s using the dig command against the FQDN", fqdnIp)
+
+	// update the IP address mapped to the FQDN in the custom DNS ConfigMap.
+	t.Logf("New IP to update to DNS | ipAdress : %s", agnHostPodTwoIP)
+	customDnsConfigMap.Data = createDnsConfig(t, map[string]string{"ipAddress": agnHostPodTwoIP, "domainName": testFQDN})
+	require.NoError(t, data.UpdateConfigMap(customDnsConfigMap), "failed to update configmap with new IP : %v", err)
+	t.Logf("successfully updated dns configMap with new IPs | ipAdress : %s", agnHostPodTwoIP)
+
+	// try to trigger an immediate refresh of the configmap by setting annotations in custom DNS server pod, this way
+	// we try to bypass the kubelet sync period which may be as long as (1 minute by default) + TTL of ConfigMaps.
+	// Ref: https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#mounted-configmaps-are-updated-automatically
+	require.NoError(t, data.setPodAnnotation(data.testNamespace, "custom-dns-server", "test.antrea.io/random-value",
+		randSeq(8)), "failed to update custom dns pod annotation.")
+
+	// finally verify that Curling the previously cached IP fails after DNS update.
+	assert.Eventually(t, func() bool {
```

Comment:
I just checked the Github workflow logs and at the moment this never fails. My understanding is that until we implement the minTTL functionality, this should eventually fail (when the entry cached by Antrea expires). What am I missing?